### PR TITLE
Update packages

### PR DESCRIPTION
**Pull request by amanmahajan7**
Fixed https://github.com/adazzle/react-data-grid/issues/1478

I am not sure if there is a better fix but setting `keep_fnames: false` fixes the problem

Similar issues
https://github.com/terser-js/terser/issues/308
https://github.com/airbnb/react-with-styles/issues/151

### CODE REVIEW

File: config/webpack.common.config.js [hunk 1]
```diff
@@ -34,19 +35,55 @@ const config = {
...
+        use: ['style-loader', 'css-loader']
       }
     ]
   },
   resolve: {
     alias: {
       common: path.resolve('packages/common/')
     }
+  },
+  optimization: {
```
Comment (amanmahajan7) [inline config/webpack.common.config.js]: Copied from https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/webpack.config.js#L186

**Comment by amanmahajan7** [comment]
Not sure why the build is failing. It passes locally 

image

File: packages/react-data-grid-examples/package.json [hunk 2]
```diff
@@ -25,9 +25,7 @@
   "devDependencies": {
     "bootstrap": "^3.2.0",
     "jquery": "^2.1.1",
-    "lodash": "^4.13.1",
-    "lodash.groupby": "^4.5.1",
     "markdown": "^0.5.0",
-    "tape": "^4.9.1"
+    "recharts": "^1.5.0"
```
Comment (MayhemYDG) [inline packages/react-data-grid-examples/package.json]: I wonder if we shouldn't just move all the deps under devDependencies for the examples.

File: packages/react-data-grid/src/__tests__/CellAction.spec.js [hunk 3]
```diff
@@ -1,104 +1,103 @@
...
-        const { wrapper } = setup({ isFirst: false });
-        const wrapperProps = wrapper.props();
-
-        expect(wrapper.length).toBe(1);
-        expect(wrapperProps.className).toBe('rdg-cell-action');
-      });
-    });
-  });
-});
+import React from 'react';
```
Comment (MayhemYDG) [inline packages/react-data-grid/src/__tests__/CellAction.spec.js]: Let's add a gitattributes file in another PR.
https://help.github.com/en/articles/dealing-with-line-endings

File: packages/react-data-grid/package.json [hunk 4]
```diff
@@ -19,5 +19,13 @@
...
     "react": "^16.0.0",
     "react-dom": "^16.0.0"
+  },
+  "devDependencies": {
+    "react-contextmenu": "^2.10.0"
+  },
+  "dependencies": {
+    "object-assign": "^4.1.1",
+    "react-is-deprecated": "^0.1.2",
+    "shallowequal": "^1.1.0"
```
Comment (MayhemYDG) [inline packages/react-data-grid/package.json]: Needs some sorting out here as well.

File: packages/react-data-grid-addons/package.json [hunk 5]
```diff
@@ -18,10 +18,21 @@
     "jquery": "^2.1.1",
     "lodash": "^4.13.1",
     "lodash.groupby": "^4.5.1",
-    "react-data-grid": "^6.0.1"
+    "moment": "^2.22.2",
+    "react-data-grid": "^6.0.1",
+    "react-dnd-test-backend": "^2.6.0",
+    "underscore": "^1.9.1"
```
[the remark this answers is not in the document]
Comment (amanmahajan7) [reply packages/react-data-grid-addons/package.json]: `moment` is marked as [external ](https://github.com/adazzle/react-data-grid/blob/692aff20942be5f1b83897d30a10ba26f694c2d8/config/webpack.common.config.js#L22)so added it as dev dependency. i do not think it is needed as the components that use it are not exported. I will remove it in a separate PR. I will move underscore

File: packages/react-data-grid-addons/package.json [hunk 6]
```diff
@@ -18,10 +18,21 @@
     "jquery": "^2.1.1",
     "lodash": "^4.13.1",
     "lodash.groupby": "^4.5.1",
-    "react-data-grid": "^6.0.1"
+    "moment": "^2.22.2",
+    "react-data-grid": "^6.0.1",
```
[the remark this answers is not in the document]
Comment (amanmahajan7) [reply packages/react-data-grid-addons/package.json]: correct

File: package-lock.json [hunk 7]
```diff
@@ -13,31 +13,89 @@
...
         "debug": "^4.1.0",
         "json5": "^2.1.0",
-        "lodash": "^4.17.10",
+        "lodash": "^4.17.11",
         "resolve": "^1.3.2",
         "semver": "^5.4.1",
         "source-map": "^0.5.0"
       },
       "dependencies": {
+        "@babel/generator": {
```
[the remark this answers is not in the document]
Comment (amanmahajan7) [reply package-lock.json]: Deleted lock files

File: packages/react-data-grid-examples/package.json (hunk 2 above)
Comment (amanmahajan7) [reply packages/react-data-grid-examples/package.json]: I think we should just delete the examples project and migrate all the examples to the new website

Comment (MayhemYDG) [reply packages/react-data-grid-examples/package.json]: Let's do that yeah.

**Pull request merged**